GLOSSARY.md

```diff
@@ -1,6 +1,50 @@
 # Glossary
 
-* **Domain Specific Language** (DSL)
-  * The functions and classes that are used by researchers to define a dataset.
-* **Query Model** (QM)
-  * The objects that are created by the DSL.
+**Backend.**
+The logical representation of a collection of EHR data.
+A Backend is associated with a provider of EHR data, a data store, and a Query Language.
+For example, the TPP Backend is associated with TPP, Microsoft SQL Server, and Transact-SQL.
+
+**Contract.**
+A specification for a collection of EHR data.
+A Contract is fulfilled by a Table.
+
+**Dataset Definition.**
+Describes, using ehrQL, the EHR data to extract from a Backend.
+
+**Domain-Specific Language (DSL).**
+A programming language designed for a specific application domain.
+For example, ehrQL is a DSL;
+it is a programming language designed for the specific application domain of EHR data.
+
+**Dummy Data.**
+Randomly-generated data used as a substitute for the EHR data to extract from a Backend.
+
+**ehrQL.**
+The Electronic Health Records Query Language, pronounced *Hercule*.
+A Domain-Specific Language (DSL) for EHR data.
+It consists of the classes, methods, and functions that are used in Dataset Definitions.
+
+**Electronic Health Records (EHR) data.**
+Data about the health of the patients in a population, stored electronically (i.e. digitally).
+
+**Measure.**
+A quotient (a numerator divided by a denominator) for a given time interval and period (e.g. each month for a year).
+
+**Query Engine.**
+Translates the Query Model into the Query Language associated with a Backend.
+For example, a Query Engine translates the Query Model into Transact-SQL, which is associated with the TPP Backend.
```

Comment:
Takes a query (a graph of connected Query Model objects), connects to a database and returns the corresponding data. Usually this will be a relational database speaking some dialect of SQL. But it doesn't have to be.